feat(excel-lists): create multilanguage json lists from excel files (DSP-1580)

irinaschubert opened the pull request:
resolves DSP-1580

irinaschubert commented:
Hallo @BalduinLandolt , nicht erschrecken, das ist ein riesiger PR. Ich habe v.a. die Funktionalität eingebaut, mehrsprachige Listen aus Excel-Files zu erstellen. Dazu musste ich eine Menge weiteren Code anpassen und habe bei dieser Gelegenheit aufgeräumt. Das Ganze ist mir etwas aus dem Ruder gelaufen, ich habe viel mehr angepasst, als für die Excel-Funktionalität nötig gewesen wäre. Und ich könnte noch lange weiter machen. Aber ich möchte diesen Task natürlich auch bald einmal abschliessen, damit die Funktionalität dem CS-Team zur Verfügung steht. 

Jedenfalls, der Kern der Sache liegt im File knora/dsplib/utils/excel_to_json_lists.py. Du hast den fast gleichen Code ja auch schon in dsp-tools-prep reviewed.

Vielen Dank schon mal!

BalduinLandolt reviewed (approved):
Some minor comments, apart from that I think it's a big improvement.

I think many of the things I noted are due to PyCharm's auto formtatting, which I'm not overly fond of... but that's cosmetics, so you can happily ignore those comments.

irinaschubert commented on knora/dsp_tools.py:
```diff
@@ -27,68 +28,72 @@ def program(args: list) -> None:
         exit(0)
 
-    if args.action == "create":
+    if args.action == 'create':
```
No, I just thought it would be better to be consistent either `'` or `"`. I chose `'` for no particular reason...

BalduinLandolt replied:
okay :)

irinaschubert commented on knora/dsplib/utils/onto_create_lists.py:
```diff
@@ -1,104 +1,101 @@
+        user: Username (e-mail) for the DSP server, has to have the permissions to create an ontology
+        password: Password of the user
+        verbose: Verbose output if True
+        dump: ???
```
I agree :)

irinaschubert commented on knora/dsplib/utils/xml_upload.py:
```diff
@@ -6,13 +6,13 @@
+from ..models.project import Project
+from ..models.resource import ResourceInstanceFactory
+from ..models.sipi import Sipi
+from ..models.value import KnoraStandoffXml
```
the `.`and `..`and also the `dsplib.`notation is only allowed if the prefix is listed in the BUILD.bazel file. But yes, we should decide which way we want it to have and then use it or leave it consistently in all files.

irinaschubert commented:
> Some minor comments, apart from that I think it's a big improvement.
> 
> I think many of the things I noted are due to PyCharm's auto formtatting, which I'm not overly fond of... but that's cosmetics, so you can happily ignore those comments.

I agree with everything and updated the files accordingly. I also changed some of my options in PyCharm in order to reformat the files nicely in the future.
